src/core/meshcop/tcat_agent.cpp

```diff
@@ -523,6 +533,52 @@ Error TcatAgent::HandleSetActiveOperationalDataset(const Message &aIncomingMessa
 
     Get<ActiveDatasetManager>().SaveLocal(dataset);
 
+    SuccessOrExit(error = otBleSecureGetPeerCertificateRaw(&GetInstance(), buf, &bufLen));
+    Get<Settings>().SaveTcatCommissionerCertificate(buf, (uint16_t)bufLen);
+exit:
+    return error;
+}
+
+Error TcatAgent::HandleGetActiveOperationalDataset(Message &aOutgoingMessage, bool &aResponse)
+{
+    Error         error = kErrorNone;
+    Dataset       dataset;
+    Dataset::Tlvs datasetTlvs;
+
+    if (!CheckCommandClassAuthorizationFlags(mCommissionerAuthorizationField.mApplicationFlags,
+                                             mDeviceAuthorizationField.mApplicationFlags, &dataset))
+    {
+        error = kErrorRejected;
+        ExitNow();
+    }
+
+    SuccessOrExit(error = Get<ActiveDatasetManager>().Read(datasetTlvs));
+    SuccessOrExit(
+        error = Tlv::AppendTlv(aOutgoingMessage, kTlvResponseWithPayload, datasetTlvs.mTlvs, datasetTlvs.mLength));
+    aResponse = true;
+
+exit:
+    return error;
+}
+
+Error TcatAgent::HandleGetCommissionerCertificate(Message &aOutgoingMessage, bool &aResponse)
+{
+    Error         error = kErrorNone;
+    Dataset       dataset;
+    unsigned char buf[1024];
+    uint16_t      bufLen = sizeof(buf);
+
+    if (!CheckCommandClassAuthorizationFlags(mCommissionerAuthorizationField.mApplicationFlags,
+                                             mDeviceAuthorizationField.mApplicationFlags, &dataset))
+    {
+        error = kErrorRejected;
+        ExitNow();
+    }
+
+    VerifyOrExit(kErrorNone == Get<Settings>().GetTcatCommissionerCertificate(buf, bufLen), error = kErrorInvalidState);
+    SuccessOrExit(error = Tlv::AppendTlv(aOutgoingMessage, kTlvResponseWithPayload, buf, bufLen));
+    aResponse = true;
+
 exit:
     return error;
 }
```

Comment:
The HandleDecomission(void) code lacks autorization checks for command class  Decommissioning (using mDecommissioningFlags).

Decommissioning shall also trigger storing the commissioner certificates.
See: https://threadgroup.atlassian.net/issues/BHC-713

Reply:
fixed